### ISSUE TITLE
Capitolo su Java

### ISSUE THREAD

**Issue by m1gwings** [m1gwings/ing-sw-cheatsheet#6]
Paragrafi da scrivere:
- [ ] ...  
- [ ] Metodi di utility
  - [ ] max 
- [ ] Enum
  - [ ] values
  - [ ] ...
- [ ] Stringhe
  - [ ] charAt  
- [ ] Array
- [ ] Collezioni 
  - [ ] Array lsit
  - [ ] Set
    - [ ] iterare sui set
  - [ ] List
    - [ ] Arrays.asList
    - [ ] get
    - [ ] add
    - [ ] clear
    - [ ] size
    - [ ] remove
    - [ ] ...
    - [ ] contains
  - [ ] Map

**Comment by FrancezcoPezce** [comment]
Aggiungerei:
-un paio di righe in cui vengono dette le implementazioni principali delle collezioni inserite (ad esempio per List ArrayList e LinkedList)
-il metodo toString() delle collezioni principali (anche in modo informale, in alcuni temi di esame chiedevano di stampare una lista e non sapevo se si andava a capo tra elementi, se e quali parentesi c'erano ecc)
-i metodi principali di Array (asList, sort, toString) e la loro creazione
-un ulteriore esempio sugli iteratori (si veda allegato): è comparso in 3 temi d'esame (in effetti erano vecchi ma comunque 3)
[iterator.pdf](https://github.com/m1gwings/ing-sw-cheatsheet/files/10335896/iterator.pdf)

Per me non è un problema occuparmi di tutte o alcune di queste cose, ma non essendo pratico di git ed essendo il capitolo scritto da qualcun altro non vorrei mettere le mani sul capitolo di qualcuno e rovinarlo.

**Comment by m1gwings** [comment]
Concordo. L'iteratore magari lo aggiungerei alla fine se abbiamo tempo: il nostro Prof ha detto durante l'ultima lezione che prima veniva rihciesto perché il programma era leggermente diverso ma ora in teoria non dovrebbero più chiederelo; noi dello scaglione E-P mi pare che non abbiamo mai visto un'implementazione completa. Il resto mi sembra tutto molto valido.

Per ora @francescogemma sta lavorando al capitolo, ed è meglio cercare di non lavorare in due sullo stesso file per evitare conflitti. Quando finisce di lavorarci, se non le ha già scritte lui, ti faccio sapere e, se ti va, puoi aggiungerle tu.

**Comment by FrancezcoPezce** [comment]
ok perfetto
